Source/Core/Core/IOS/ES/TitleManagement.cpp

```diff
@@ -644,39 +654,40 @@ IPCCommandResult ES::ExportContentBegin(Context& context, const IOCtlVRequest& r
 
 ReturnCode ES::ExportContentData(Context& context, u32 content_fd, u8* data, u32 data_size)
 {
-  const auto iterator = context.title_export.contents.find(content_fd);
-  if (!context.title_export.valid || iterator == context.title_export.contents.end() ||
-      iterator->second.content.m_position >= iterator->second.content.m_content.size)
+  ReturnCode ret = ES_EINVAL;
+  if (context.title_import.valid && context.title_import.content.valid && data && data_size != 0)
   {
-    return ES_EINVAL;
-  }
-
-  auto& metadata = iterator->second.content;
-
-  const auto& content_loader = AccessContentDevice(metadata.m_title_id);
-  const auto* content = content_loader.GetContentByID(metadata.m_content.id);
-  content->m_Data->Open();
-
-  const u32 length =
-      std::min(static_cast<u32>(metadata.m_content.size - metadata.m_position), data_size);
-  std::vector<u8> buffer(length);
-
-  if (!content->m_Data->GetRange(metadata.m_position, length, buffer.data()))
-  {
-    ERROR_LOG(IOS_ES, "ExportContentData: ES_SHORT_READ");
-    return ES_SHORT_READ;
+    ret = [&] {
```

Comment:
I had a hard time following the flow of this function because of the lambda usage. IMO deduplicating the `CloseContent(content_fd, 0); context.title_import = {};` code isn't worth the hit to readability the lambda introduces.